src/snowcli/cli/snowpark/commands.py

```diff
@@ -143,6 +144,22 @@ def deploy(
             "Please use build command to create it."
         )
 
+    pm = ProcedureManager()
+    fm = FunctionManager()
+    om = ObjectManager()
+
+    _check_if_all_defined_integrations_exists(om, functions, procedures)
+
+    existing_functions = _check_if_any_object_exist_already(fm, functions)
+    existing_procedures = _check_if_any_object_exist_already(pm, procedures)
+
+    if (existing_functions or existing_procedures) and not replace:
+        msg = "Following objects already exists. Consider using --replace.\n"
```

Comment:
Do you think it will be enough to test it only using unit tests? We rely here a lot on matching of our data with Snowflake data. Maybe this case deserves an integration test? Wdyt?

Reply:
Will adjust integration test